dali/server/daldap.cpp

```diff
@@ -148,8 +148,8 @@ class CDaliLdapConnection: implements IDaliLdapConnection, public CInterface
         Owned<ISecUser> user = ldapsecurity->createUser(username);
         user->setAuthenticateStatus(AS_AUTHENTICATED);
 
-        bool filescope = stricmp(key,"Scope")==0;
-        bool wuscope = stricmp(key,"workunit")==0;
+        bool filescope = key && stricmp(key,"Scope")==0;
```

Comment:
use strieq perhaps?

Reply:
(perhaps strieq should be made safe!)

Reply:
possibly, but I was aiming for minimal changes.  Not convinced about strieq.